
```{r setup, include=FALSE}
setwd("<INSERT WORKING DIRECTORY>")
root = getwd()
knitr::opts_chunk$set(root = root)
knitr::opts_knit$set(root = root)
```

Load libraries
```{r}
library(data.table)
library(stringr)
```

New files
```{r}
ccle = fread(paste0(root, "/CCLE_expression.csv"))
ceres = fread(paste0(root, "/CRISPR_gene_effect.csv"))
mutations = fread(paste0(root, "/CCLE_mutations_bool_damaging.csv"))
proteomics = fread(paste0(root, "/Proteomics.csv"))
methylation = fread(paste0(root, "/Methylation_(1kb_upstream_TSS).csv"))
```

Expression vs dependency
```{r}
anchor_names_expression = c("BRCA1 (672)", "MLH1 (4292)", "BCL2L1 (598)", "VPS4A (27183)")
target_names_dependency = c("PARP1 (142)", "WRN (7486)", "MCL1 (4170)", "VPS4B (9525)")

setnames(ccle, "V1", "depmap_id")
setnames(ceres, "DepMap_ID", "depmap_id")

get_rank = function(i, omic_matrix, dependency_matrix, anchors, targets, order_decreasing = TRUE, samples = NA) {
   # gene we are working on
  curgene = anchors[i]
  # make matrix with expression of anchor and dependency of everything else
  gene_vs_all = merge(omic_matrix[, .(depmap_id, get(curgene))], dependency_matrix, by = "depmap_id")

  # set appropriate colname, df instead of dt, set rownames for correlation
  setnames(gene_vs_all, "V2", curgene)
  gene_vs_all = as.data.frame(gene_vs_all)
  rownames(gene_vs_all) = gene_vs_all$depmap_id
  gene_vs_all$depmap_id=NULL
  
  # get correlation of anchor expression with dependency of everything else
  gene_cor = as.data.frame(cor(gene_vs_all[,-1], gene_vs_all[,1], use = "pairwise.complete.obs"))
  
  # get rank of target gene
  gene_cor = gene_cor[order(gene_cor$V1, decreasing = order_decreasing),, drop = F]
  rank = which(rownames(gene_cor)==targets[i])
  total_length = nrow(gene_cor)
  return(list(rank = rank, total_length = total_length))
}

lapply(1:4, get_rank, ccle, ceres, anchor_names_expression, target_names_dependency)
```

Proteomics vs dependency
```{r}
setnames(proteomics, "V1", "depmap_id")

anchor_names_proteomics = c("BRCA1 (P38398)", "MLH1 (P40692)", "BCL2L1 (Q07817)", "VPS4A (Q9UN37)")

lapply(1:4, get_rank, proteomics, ceres, anchor_names_proteomics, target_names_dependency)
```

For methylation and mutation the ranking is reversed because more mutation or methylation corresponds with less gene activity (whereas more RNA or protein corresponds with more gene activity).

Mutation vs dependency
```{r}
setnames(mutations, "V1", "depmap_id")
colnames(mutations)[stringr::str_detect(colnames(mutations), "BRCA1|MLH1|BCL2L1|VPS4A")]

# select proper names for the anchors
anchor_names_mutations = c("BRCA1 (672)", "MLH1 (4292)", "BCL2L1 (598)", "VPS4A (27183)")

lapply(1:4, get_rank, mutations, ceres, anchor_names_mutations, target_names_dependency, order_decreasing = F)
```

Methylation vs dependency
```{r}
# for reference
target_names_dependency = c("PARP1 (142)",
                            "WRN (7486)", 
                            "MCL1 (4170)", 
                            "VPS4B (9525)")

# now run the actual ranking-- a special version of the other function where we take the mean of all proteomic measurements per anchor
setnames(methylation, "V1", "depmap_id")

anchors = c("BRCA1", "MLH1", "BCL2L1", "VPS4A")

get_rank_methylation = function(i, omic_matrix, dependency_matrix, anchors, targets, order_decreasing = TRUE, samples = NA) {
  # gene we are working on
  curgene = anchors[i]
  # colnames of all columns for anchor genes
  selectcols = colnames(methylation)[grep(paste0(curgene, "_"), colnames(methylation))]
  
  # make matrix with expression of anchor and dependency of everything else
  tmp = methylation[, c("depmap_id", colnames(methylation)[grep(paste0(curgene, "_"), colnames(methylation))]), with = F]
  tmp[, anchormeans := rowMeans(.SD, na.rm = T), .SDcols = selectcols]
  gene_vs_all = merge(tmp[, .(depmap_id, anchormeans)], ceres, by = "depmap_id")
  
  # set appropriate colname, df instead of dt, set rownames for correlation
  setnames(gene_vs_all, "anchormeans", curgene)
  gene_vs_all = as.data.frame(gene_vs_all)
  rownames(gene_vs_all) = gene_vs_all$depmap_id
  gene_vs_all$depmap_id=NULL
  
  # get correlation of anchor expression with dependency of everything else
  gene_cor = as.data.frame(cor(gene_vs_all[,-1], gene_vs_all[,1], use = "pairwise.complete.obs"))
  
  # get rank of target gene
  gene_cor = gene_cor[order(gene_cor$V1, decreasing = order_decreasing),, drop = F]
  rank = which(rownames(gene_cor)==targets[i])
  total_length = nrow(gene_cor)
  return(list(rank = rank, total_length = total_length))
}

lapply(1:length(anchors), get_rank_methylation, methylation, ceres, anchors, target_names_dependency, order_decreasing = F)
```

```{r}
sessionInfo()
```

